### PR TITLE
clarification the document of --resolve flag in name.publish

### LINKED ISSUE #5368: Clarification needed for name.publish --resolve flag
#### Version information:
go-ipfs version: 0.4.17-dfd19c4
Repo version: 7
System version: amd64/darwin
Golang version: go1.10.3

#### Type:
Bug

#### Description:
`--resolve` doesn't seem to be working...

```sh
$ ipfs resolve /ipns/ipfs.io
/ipfs/QmYNQJoKGNHTpPxCBPh9KkDpaExgd2duMa3aF6ytMpHdao
$ ipfs name publish --resolve=true /ipns/ipfs.io
Published to QmahispzgXbQ5ewaxSjdrst8KNLzREhVHdhxZwAUHQWqLb: /ipns/ipfs.io
$ ipfs name resolve
Error: could not resolve name (recursion limit exceeded)
```

I was expecting:

```sh
ipfs name publish --resolve=true /ipns/ipfs.io
Published to QmahispzgXbQ5ewaxSjdrst8KNLzREhVHdhxZwAUHQWqLb: /ipfs/QmYNQJoKGNHTpPxCBPh9KkDpaExgd2duMa3aF6ytMpHdao
```

FYI, this works...:

```
$ ipfs name resolve --recursive=true
/ipfs/QmYNQJoKGNHTpPxCBPh9KkDpaExgd2duMa3aF6ytMpHdao
```

### PR DESCRIPTION
**Pull request by kjzz**
fix #5368 

License: MIT
Signed-off-by: Kejie Zhang <601172892@qq.com>

### CODE REVIEW

File: core/commands/name/publish.go [hunk 1]
```diff
@@ -73,7 +73,7 @@ Alternatively, publish an <ipfs-path> using a valid PeerID (as listed by
 		cmdkit.StringArg(ipfsPathOptionName, true, false, "ipfs path of the object to be published.").EnableStdin(),
 	},
 	Options: []cmdkit.Option{
-		cmdkit.BoolOption(resolveOptionName, "Resolve given path before publishing.").WithDefault(true),
+		cmdkit.BoolOption(resolveOptionName, "Check the given path whether can be resolved before publishing. <<default>>").WithDefault(true),
```
Comment (Stebalien) [inline core/commands/name/publish.go]: "Check if the given path can be resolved before publishing"

**Comment by kjzz** [comment]
Update.

File: core/commands/name/publish.go [hunk 2]
```diff
@@ -73,7 +73,7 @@ Alternatively, publish an <ipfs-path> using a valid PeerID (as listed by
 		cmdkit.StringArg(ipfsPathOptionName, true, false, "ipfs path of the object to be published.").EnableStdin(),
 	},
 	Options: []cmdkit.Option{
-		cmdkit.BoolOption(resolveOptionName, "Resolve given path before publishing.").WithDefault(true),
+		cmdkit.BoolOption(resolveOptionName, "Check if the given path can be resolved before publishing. <<default>>").WithDefault(true),
```
Comment (magik6k) [inline core/commands/name/publish.go]: Don't need the '<<default>>', since the cmds lib appends this automatically

Comment (Stebalien) [inline core/commands/name/publish.go]:
```suggestion
		cmdkit.BoolOption(resolveOptionName, "Check if the given path can be resolved before publishing.").WithDefault(true),
```

**Pull request merged**